src/mscorlib/src/System/Diagnostics/Eventing/EventPipe.cs

```diff
@@ -0,0 +1,155 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Security;
+using Microsoft.Win32;
+
+namespace System.Diagnostics.Tracing
+{
+    [StructLayout(LayoutKind.Sequential)]
+    internal struct EventPipeProviderConfiguration
```

Comment:
I tried making this a class, but interop doesn't seem to know how to marshal it.  That's why I originally made it a struct.  If there is a way to do this, let me know, and I can make the change.